### PR TITLE
removed default config constructors from informer

### PR DESCRIPTION
**Pull request by xh3b4sd**

### CODE REVIEW

File: informer/informer.go [hunk 1]
```diff
@@ -40,12 +40,8 @@ const (
 
 // Config represents the configuration used to create a new Informer.
 type Config struct {
-	// Dependencies.
-
 	Watcher Watcher
 
-	// Settings.
-
```
Comment (xh3b4sd) [inline informer/informer.go]: I think we do not need this. Should be clear by now how we structure this.

File: informer/informer.go [hunk 2]
```diff
@@ -57,20 +53,6 @@ type Config struct {
...
-// DefaultConfig provides a default configuration to create a new by best
-// effort.
-func DefaultConfig() Config {
-	return Config{
-		// Dependencies.
-		Watcher: nil,
-
-		// Settings.
-		ListOptions:  metav1.ListOptions{},
-		RateWait:     DefaultRateWait,
```
Comment (xh3b4sd) [inline informer/informer.go]: I think the rate wait can actually be zero, but then we do not know how to default it to something else. Right now the default is one second and I would like to keep this but the defaulting gets broken like with booleans now. Anybody a cool idea how to do this? I only have using pointers in mind.

Comment (kopiczko) [reply informer/informer.go]: Why not default to DefaultRateWait like it was?

Comment (xh3b4sd) [reply informer/informer.go]: The thing is you can just configure 0. This means there is then no rate wait. This means 0 is not the default. Based on what should we default? We only get input the user manipulated or not. We do not know if they did so. This is why the only reliable option I have in mind are pointers.

Comment (kopiczko) [reply informer/informer.go]: Ack.

**Comment by xh3b4sd** [comment]
It is nice of all of you to approve this but the problem I am talking about is not solved and I would like to get feedback on this. :D

**Comment by kopiczko** [comment]
I think defaulting to 0 and having `DefaultRateWait` is ok.

**Comment by xh3b4sd** [comment]
So you mean it is ok to require the user to use `informer.DefaultRateWait` on their own if they want to use that, which is kind of an API change.

**Comment by kopiczko** [comment]
Yes, removing `DefaultConfig` is also an API change.

**Comment by xh3b4sd** [comment]
Yes, but one change is explicit and the compiler tells you about it and the other is implicit and you can just fall for it.

**Pull request merged**